lib/graph/mod.rs

```diff
@@ -314,6 +316,19 @@ where
             }
         }
 
+        // Special handling for the start node as it can be part of a loop.
+        // This is necessary because we don't have a dedicated entry node.
+        for edge in &self.edges_in[&start_index] {
+            let mut runner = edge.head();
+            loop {
+                df.get_mut(&runner).unwrap().insert(start_index);
+                if !idoms.contains_key(&runner) {
+                    break;
+                }
+                runner = idoms[&runner];
+            }
+        }
```

Comment:
See test case `test_dominance_frontiers_of_graph_with_start_node_in_loop` which required this special handling.

Reply:
Another more elegant but also more expensive approach would be to temporarily clone the graph, add a new dedicated entry node and connect it to the start node. This would avoid the "special handling" at the cost of a clone.